### PR TITLE
PWN-7169 - widget click actions

### PR DESCRIPTION
**Pull request by NikolaiEmelianov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7169

## Description of Work

widget click actions

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapFragment.kt [hunk 1]
```diff
@@ -55,12 +55,18 @@ class JupiterSwapFragment :
         super.onViewCreated(view, savedInstanceState)
         with(binding) {
             swapWidgetFrom.onAmountChanged = { presenter.onTokenAmountChange(it) }
+            swapWidgetFrom.onAllAmountClick = { presenter.onAllAmountClick() }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapFragment.kt:58]: maybe we can do method reference for all callbacks?

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapPresenter.kt [hunk 2]
```diff
@@ -60,6 +61,60 @@ class JupiterSwapPresenter(
         stateManager.onNewAction(SwapStateAction.SwapSuccess)
     }
 
+    override fun onAllAmountClick() {
+        val allTokenAAmount = when (val featureState = featureState) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapPresenter.kt:65]: maybe we can extract all this when logic to interactor?

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapPresenter.kt [hunk 3]
```diff
@@ -60,6 +61,60 @@ class JupiterSwapPresenter(
...
+            SwapState.InitialLoading,
+            is SwapState.SwapException.FeatureExceptionWrapper,
+            is SwapState.SwapException.OtherException -> null
+        }
+        return (tokenA as? SwapTokenModel.UserToken)?.details?.total
+    }
+
+    override fun onChangeTokenAClick() {
+        if (isPossibleOpenChangTokenScreen(featureState))
+            view?.openChangeTokenAScreen()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapPresenter.kt:95]: {}

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapPresenter.kt [hunk 4]
```diff
@@ -60,6 +61,60 @@ class JupiterSwapPresenter(
...
+            is SwapState.TokenAZero -> state.tokenA
+            SwapState.InitialLoading,
+            is SwapState.SwapException.FeatureExceptionWrapper,
+            is SwapState.SwapException.OtherException -> null
+        }
+        return (tokenA as? SwapTokenModel.UserToken)?.details?.total
+    }
+
+    override fun onChangeTokenAClick() {
+        if (isPossibleOpenChangTokenScreen(featureState))
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapPresenter.kt:94]: isChangeTokenScreenAvailable

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapPresenter.kt [hunk 5]
```diff
@@ -60,6 +61,60 @@ class JupiterSwapPresenter(
...
+            SwapState.InitialLoading -> false
+            is SwapState.LoadingRoutes,
+            is SwapState.LoadingTransaction,
+            is SwapState.SwapLoaded,
+            is SwapState.TokenAZero -> true
+            is SwapState.SwapException.FeatureExceptionWrapper ->
+                isPossibleOpenChangTokenScreen(featureState.previousFeatureState)
+            is SwapState.SwapException.OtherException ->
+                isPossibleOpenChangTokenScreen(featureState.previousFeatureState)
+        }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapPresenter.kt:115]:
```suggestion
        return when (featureState) {
            null,
            SwapState.InitialLoading -> false
            is SwapState.LoadingRoutes,
            is SwapState.LoadingTransaction,
            is SwapState.SwapLoaded,
            is SwapState.TokenAZero -> true
            is SwapState.SwapException ->
                isPossibleOpenChangTokenScreen(featureState.previousFeatureState)
        }
```

**Review by gslevinkov — APPROVED** [review]

File: buildSrc/src/main/java/AppVersions.kt [hunk 6]
```diff
@@ -3,5 +3,5 @@ object AppVersions {
     const val VERSION_MINOR = 4
     const val VERSION_HOTFIX = 1
     const val VERSION_PATCH = 0
-    const val VERSION_BUILD = 1677077189612
```
Comment (kabramovp2p) [inline buildSrc/src/main/java/AppVersions.kt:6]: ?

File: buildSrc/src/main/java/LastCommit.kt [hunk 7]
```diff
@@ -1,3 +1,3 @@
 object LastCommit {
-    const val COMMIT_HASH = "fa04ce791639e39883a2fe7afe7d5a1865231aec"
```
Comment (kabramovp2p) [inline buildSrc/src/main/java/LastCommit.kt:2]: ?
i think all this should be reverted

**Review by kabramovp2p — APPROVED** [review]

**Pull request merged**